overlord/snapstate/component.go

```diff
@@ -37,9 +38,19 @@ import (
 // InstallComponents installs all of the components in the given names list. The
 // snap represented by info must already be installed, and all of the components
 // in names should not be installed prior to calling this function.
-//
-// TODO:COMPS: respect the transaction that is passed to this function
 func InstallComponents(ctx context.Context, st *state.State, names []string, info *snap.Info, opts Options) ([]*state.TaskSet, error) {
+	if opts.Flags.Transaction != client.TransactionAllSnaps && opts.Flags.Lane != 0 {
+		return nil, errors.New("cannot specify a lane without setting transaction to \"all-snaps\"")
+	}
+
+	if opts.Flags.Transaction == client.TransactionAllSnaps && opts.Flags.Lane == 0 {
+		opts.Flags.Lane = st.NewLane()
+	}
+
+	if err := setDefaultSnapstateOptions(st, &opts); err != nil {
+		return nil, err
+	}
```

Comment:
we probably need to check the DeviceCtx behavior of this? atm if I remove it nothing breaks in the tests

Reply:
Added some cases in 2c7bd89e62882d43e8974a82cfbdf39fe29df526.